public/app/features/search/page/components/columns.tsx

```diff
@@ -442,3 +498,35 @@ function getDisplayValue({
   }
   return formattedValueToString(getDisplay(value));
 }
+
+/**
+ * Calculates the rough duration between two dates, keeping only the most significant unit e.g:
```

Comment:
okay - @ashharrison90 @L-M-K-B @Clarity-89 got a new implementation here. Haven't actually tested it yet though.

Thoughts?

Reply:
since they're all optional, you could type the return type of the function as `Intl.DurationInput` and do something like:
```tsx
  const { years = 0, months = 0, days = 0, hours = 0, minutes = 0 } = intervalToDuration({ start, end });

  if (years > 0) {
    return { years };
  } else if (months > 0) {
    return { months };
  } else if (days > 0) {
    return { days };
  } else if (hours > 0) {
    return { hours };
  } else {
    return { minutes };
  }
```

tbh i'm not sure which is clearer 🤔 

my only other qualm is with the name. think you want `calcCoarseDuration` (a instead of u), but `calcRoughDuration` or `calcRoundedDuration` also sound fine to me 🤷

Reply:
good catch on both parts!